fix: DIA-2071: [BE,FE] AttributeError: 'AnonymousUser' object has no attribute 'active_organization'

nick-skriabin opened the pull request:
Adding proper 401 handling on BE and FE to redirect users to login.

This pull request includes changes to the `APIProxy` class to introduce a new callback function that is triggered when a request is finished. This enhancement allows for additional handling or processing after a request completes. 

>[!NOTE]
> **Context**
> 
> We're starting to heavily use `atomWithQuery` where `useAPI` is not available essentially loosing the ability to properly handle 401 and redirect to the login screen
> This PR addresses that by adding a callback to the API that will run right after the request is finished. The idea is to handle 401 on the APIProxy initialization level.
> With this PR and use of API in the `atomWithQuery` directly we can correctly handle 401 and the redirect.

Enhancements to `APIProxy` class:

* [`web/libs/core/src/lib/api-proxy/index.ts`](diffhunk://#diff-1d528c3a0818ed06532cc9c5c51090d71240fe04fa3c82edfd9ae108c6bd0287R44-R45): Added a new optional callback function `onRequestFinished` in the `APIProxy` class that is called with the response when a request is finished. [[1]](diffhunk://#diff-1d528c3a0818ed06532cc9c5c51090d71240fe04fa3c82edfd9ae108c6bd0287R44-R45) [[2]](diffhunk://#diff-1d528c3a0818ed06532cc9c5c51090d71240fe04fa3c82edfd9ae108c6bd0287R54) [[3]](diffhunk://#diff-1d528c3a0818ed06532cc9c5c51090d71240fe04fa3c82edfd9ae108c6bd0287R238)
* [`web/libs/core/src/lib/api-proxy/types.ts`](diffhunk://#diff-6fbec4773da55113ef066e07975569b15554e80fa7f41c5e15a80a5bc1491d51R32): Updated the `APIProxyOptions` type to include the new `onRequestFinished` optional callback function.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
Add mid-request event handler

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2071/unauthorized-access

nick-skriabin committed:
Add 401 handling

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14265698013)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-2071/unauthorized-access'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14265698013

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14265698013

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14302203105)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-2071/unauthorized-access'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14302203105

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14302203105
